src/engraving/dom/chord.cpp

```diff
@@ -3267,14 +3267,16 @@ void GraceNotesGroup::setPos(double x, double y)
 void GraceNotesGroup::addToShape()
 {
     for (Chord* grace : *this) {
+        const StaffType* st = grace->staffType();
+        const PointF yOffset = PointF(0.0, st ? st->yoffset().val() * grace->spatium() : 0.0);
```

Comment:
These two lines of code are repeated dozens of times, I won't comment on all of them. They should all become a method of EngravingItem (see my other comment).